src/main/java/com/mg105/entities/Inventory.java

```diff
@@ -0,0 +1,61 @@
+package com.mg105.entities;
+import java.util.ArrayList;
+
+public class Inventory {
+    final static int inventoryLimit = 10;
+    private static ArrayList<Item> items = new ArrayList<>();
+
+    public static int getInventoryLimit(){
+        return inventoryLimit;
+    }
+
+    public static int getNumberOfItems(){
+        return items.size();
+    }
+
+    public static boolean isInventoryFull(){
+        return getNumberOfItems() >= getInventoryLimit();
+    }
+
+    /**
+     * Precondition: Inventory is not full
+     * If precondition is violated, the function throws a RuntimeException
+     * */
+    public static void addItem(Item item) throws RuntimeException{
+        if(isInventoryFull()){
+            throw new RuntimeException("Item cannot be added when Inventory is full");
+        }
```

Comment:
If we ever want to catch this (which we probably do to show some message that the inventory is full), .

Side note: in general exceptions are for exceptional circumstances, does this really fit the bill (IMO it seems very likely that a user will get into a situation where we have a full inventory).

Side note 2: If you *really* need this check, `assert` statements are probably better.

Reply:
The reason this function throws here is that interactor is so post of check before hand if the inventory if full and only add the item if it is not, so this case should be exception IMO. Also in lecture the prof said to throw when a precondition is violated.